### DISCUSSION TITLE
Handy additions for interactive mode

### DISCUSSION THREAD

**Discussion by mcseemz** [spring-projects/spring-shell#471] [Ideas]
Hey, respect to developers.

Spring Shell in interactive mode turned out to be handy for text-based games. But there are two cases that make things more complex than desired:

1. Component Flow for multiple item selector  (`.withMultipleItemSelector`) cannot control the initial selection state for `selectItems`. `selectItems` always appear deselected for the user.
2. It is not possible to have command already suggested in the shell input. e.g. `shell:>start`, with further control over suggestions depending on the current application state.

If that can be addressed, would make life a little easier.
Thanks!

**Comment by jvalkeal** [comment]
1. It's worked on pr #468
2. As we're first completing commands(those are known from command registry) and then command options(which can be customised).

Not sure about programmatic suggestion without user interaction as I hear about an idea first time. Depends if jline has some features around it.

**Comment by mcseemz** [comment]
Hi Janne
1.  great, actually.
2. If I as a user can enter random text into the console before hitting Enter, would be nice to do the same programmatically. Looking at https://github.com/jline/jline3/blob/master/reader/src/main/java/org/jline/reader/LineReader.java#L634 , they operate a buffer with the default value, so that maybe even possible.

And another couple of issues that popped up in the game dev context.
3. TableBuilder considers ANSI control chars as physical while calculating cell width. So it is not possible to use any JLine color within the cells.
4. default Confirmation component color is white for a text for some reason (unreadable on terminals with white background)

Regards,
Maxim

**Reply by jvalkeal** [reply]
`TableBuilder` probably goes through overhaul in `3.x` as there's various issues as you noticed.

Good catch for component styling as I see same now that I tried it. We'll fix it. #476
